xcube/core/gen/iproc.py

```diff
@@ -260,6 +271,24 @@ def pre_process(self, dataset: xr.Dataset) -> xr.Dataset:
 
         return _normalize_lon_360(dataset)
 
+    def get_spatial_subset(self, dataset: xr.Dataset, dst_region: Tuple[float, float, float, float]) -> xr.Dataset:
+        lon_min, lat_min, lon_max, lat_max = dst_region
```

Comment:
First make a test, if `dataset` bounds are entirely included in `dst_region`. If so, return `dataset` unchanged:

```python
from xcube.core.geom import get_dataset_geometry
from xcube.core.geom import get_box_split_bounds_geometry

ds_bounds_geom = get_dataset_geometry(dataset)
dst_region_geom = get_box_split_bounds_geometry(*dst_region)
if dst_region_geom.contains(ds_bounds_geom):
    return dataset
```

Reply:
Does ``` dataset_subset.lon```work when the coord_name is 'x'?

Reply:
Yes, and it expected to be 2-D.